text/trace/0189-instrumentation-layers-suppression.md

```diff
@@ -0,0 +1,222 @@
+# Instrumentation Layers and Suppression
+
+This document describes approach for tracing instrumentation layers and suppressing duplicate layers.
+
+## Motivation
+
+- Provide clarity for instrumentation layers: e.g. DB calls on top of REST API

```

Comment:
Related to the questions about a definition of layers: there are discussing about actually modelling HTTP request with multiple spans (for e. g. retries or forwards). Should we aim to cover those cases with a suppression proposal?

Reply:
nope, for now, I assume we have one layer of HTTP spans (if we'll have more for logical + physical, I'd treat them as two different layers) - they can't carry the same information when both are enabled.

HTTP instrumentation would need configuration to emit 1) logical or 2) physical or 3) both to avoid redundancy and doubling of costs. I don't see a way to achieve it with suppression